distro/install_script/default.nix

```diff
@@ -0,0 +1,38 @@
+{ disable-systemd ? "false", stage-2-hook ? "/bin/sh -l", log-level ? "error"
+, console ? "hvc0", test-command ? "", pkgs ? import <nixpkgs> { } }:
+let
+  aster-kernel = builtins.path {
+    path = ../../target/osdk/iso_root/boot/aster-nix-osdk-bin;
+  };
+  distro = builtins.path { path = ../../distro; };
+
+  aster_configuration = pkgs.replaceVarsWith {
+    src = ./aster_configuration.nix;
+    replacements = {
+      aster-kernel = aster-kernel;
+      aster-disable-systemd = disable-systemd;
+      aster-stage-2-hook = stage-2-hook;
+      aster-log-level = log-level;
+      aster-console = console;
+      aster-test-command = test-command;
+    };
+  };
+  install_nixos = pkgs.replaceVarsWith {
+    src = ./install_nixos.sh.template;
+    replacements = {
+      aster-configuration = aster_configuration;
+      aster-distro = distro;
+      aster-kernel = aster-kernel;
+    };
+    isExecutable = true;
+  };
+
+in pkgs.stdenv.mkDerivation {
+  name = "nixos_install_script";
```

Comment:
The package will be visible to the entire to a greater audience than the Asterinas developers. So the name should include Asterinas. For example, you can name it `aster_nixos_installer`.